### DISCUSSION TITLE
Is anyone using "IsControllerNameValid" util function?

### DISCUSSION THREAD

**Discussion by robscott** [kubernetes-sigs/gateway-api#2594] [General]
We have a util function for each API version in `/apis` for this, but I don't really know _why_ we need this. Would be helpful to hear if anyone is relying on it, otherwise I think we should just remove it.

https://github.com/kubernetes-sigs/gateway-api/blob/8426b7d44f21735a57f42db65480b6e817acdeec/apis/v1/util/validation/gatewayclass.go#L29-L34

**Comment by youngnick** [comment]
I think the original intent was to give folks a way to validate against the rules that will be applied - I think that with better validation and CEL, we can be more confident that invalid values won't be in there.

**Comment by sunjayBhatia** [comment]
Contour is not using this helper
